### DISCUSSION TITLE
Problemi ad istanziare HA tramite Docker compose

### DISCUSSION THREAD

**Discussion by braudoo** [indomus/forum#1924] [HUB: Home Assistant]
Salve a tutti, vorrei installare HA tramite Docker-Compose, ma dopo aver seguito tutta la procedure, essermi messo nella cartella contenente il file **docker-compose.yaml** e aver lanciato
`docker-compose up -d home-assistant`
questo mi restituiva:
`-bash: docker-compose: command not found`

Le verifiche sulla versione di Docker, Docker-compose e libseccomp sono tutte con le ultime versioni.

Cosa può essere?

Grazie.

**Comment by snakuzzo** [comment]
Scusa...ma se ti dà "command not found" come fai a verificare la versione di docker-compose?

**Comment by braudoo** [comment]
purtroppo non lo so. 
Ho provveduto ad istanziare HA tramite il comando, con successo.
Ho voluto provare tramite Docker-compose ad istanziare **mosquitto** ma anche in questo caso niente:

```
pi@raspberrypi:/home/emanuele/compose $ docker compose version
Docker Compose version v2.10.2
pi@raspberrypi:/home/emanuele/compose $ sudo docker run hello-world

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (arm64v8)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
 https://hub.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/get-started/

pi@raspberrypi:/home/emanuele/compose $ sudo docker-compose up -d mosquitto
sudo: docker-compose: command not found
pi@raspberrypi:/home/emanuele/compose $ docker-compose up -d mosquitto
-bash: docker-compose: command not found
```

Anche in questo caso non sono riuscito.

Cosa sto sbagliando?

Grazie

**Reply by snakuzzo** [reply]
Stai sbagliando tutto. Stai facendo un casino
Il primo comando che hai lanciato è il compose command di docker, non è `docker-compose`. Leggi [QUI](https://docs.docker.com/compose/cli-command-compatibility/)!!!

Non hai installato `docker-compose`...per questo non lo trova.
Inoltre, non andare a caso usando `sudo` a volte sì e a volte no. Devi capire cosa significa usare `sudo`.

**Reply by braudoo** [reply]
Come mai dici che Docker-compose non è installato se poi lanciando il comando `docker compose version` mi dà la versione?
Riguardo la guida in merito all installazione di Docker-compose e vedo cosa mi sono perso.

**Reply by snakuzzo** [reply]
Perché

```
docker compose
```

non è la stessa cosa di

```
docker-compose
```

**Reply by snakuzzo** [reply]
E visto che mi chiedi perché te l'ho detto, credo tu non abbia letto il link che ti ho passato ieri

**Reply by braudoo** [reply]
No, l'ho provato a leggere ma faccio fatica a capirlo.
Sto cercando di capire sul sito di **docker.com** quali passaggi ho mancato o non sto facendo.
Come procedura ho utilizzato quello indicato in **Install using the repository**  ma non ho fatto il set-up di questo.
Presumo che sia questo il passaggio che ho mancato.

**Reply by indomus** [reply]
abbi pazienza, ma cosa c'è da capire? Diego ti ha scritto chiaramente di usare il comando corretto, ovvero `docker-compose`, mentre tu usi `docker compose`.

**Reply by braudoo** [reply]
Grazie per la pazienza sono verginello in questo campo.
Ho capito ora che tra **Compose** (inteso come Docker Compose) e **Docker-compose** c'è differenza. 
Quindi in sostanza NON ho installato il plugin di docker-compose, ma solo il Compose, quindi per avviare lo stack (in questo caso quello di **mosquitto**) devo fare semplicemente
`docker compose up -d mosquitto `

Ne ho beccata qualcuno?

**Reply by indomus** [reply]
`docker-compose`

**DOCKER-COMPOSE**
